### ISSUE TITLE
Prevent Page break after title

### ISSUE THREAD

**Issue by plaindocs** [asciidoctor/asciidoctor-pdf#105]
When printing a simple 1 page piece of text to PDF, how can I prevent the page break after the document title?

**Comment by plaindocs** [comment]
Obviously using == instead of = as the top level header works, but seems kludgy.

**Comment by mojavelinux** [comment]
Another option is to use a discrete level-0 heading. I remember posting that advice in a similar issue, so I know I've tested it and it works out. See #95.

Still, this is clearly something we'd like to have control over.

**Comment by plaindocs** [comment]
Thanks. To be honest that seems even more kludgy. (even if it works better :-p)

Discussions about defaults for article/book aside, it should 'obviously' be a theme/formatting thing, not a content wise header level thing.

**Comment by mojavelinux** [comment]
> it should 'obviously' be a theme/formatting thing, not a content wise header level thing.

Sounds reasonable. We might even be able to hook it to an attribute. I'll give it some thought. If I don't think of anything else, we'll go with the theme setting.

**Comment by dcp65** [comment]
Any update on this issue? 

I have a PDF with a number of orphaned second and third-level headers at the bottom of pages. I'd like to know how to work around this without forcing a page break in the source.

**Comment by dmitrij-drandarov** [comment]
Any update yet?

**Comment by das-g** [comment]
> Obviously using == instead of = as the top level header works, but seems kludgy.

It's AFAIK also semantically wrong to use `==` there if you actually want your document to have a title (rather than the first section having that title.)

The problems you noticed are probable because a Level-1 title cannot start the document header. (So the stuff there is regarded as regular content.)

**Comment by ogmios-voice** [comment]
I have very little ruby experience, but this quick fix **turns off** the page break after the title page based on the `title_page.no_break_after` (new) property of a theme:

You just need to extend a single line in `converter.rb` in the `convert_document` method:

``` ruby
layout_title_page doc

# NOTE a new page will already be started if the cover image is a PDF
- start_new_page unless page_is_empty?
+ start_new_page unless page_is_empty? || @theme.title_page_no_break_after
```

theme example (part):

``` yml
title_page:
  no_break_after: yes
```

The tests were done with `asciidoctor-pdf-1.5.0.alpha.16`.

**Comment by mojavelinux** [comment]
I appreciate that contribution. However, in my mind, an article doctype shouldn't have a title page. A title page is reserved for the book doctype. So the proper fix is to simply not create a title page for an article.

We could still add a switch that would preserve the existing behavior for those who rely on it. But the default should be to put the doctitle at the top of the first page when the doctype is article.

**Comment by mojavelinux** [comment]
I've submitted a proposal. The behavior can be reversed by setting the `title-page` attribute in the document header.

The only thing I wasn't able to do is put the TOC on the same page. The logic for the TOC is very complicated and assumes it begins and ends on a dedicated page, so I'm afraid to touch it. For now, the TOC shows before the first page even though the correct place for it is immediately below the doctitle.

**Comment by ogmios-voice** [comment]
I can try allocating some time and do the solution you would prefer. But for this I would have a few questions first:

1. Do you prefer extending the `layout_title_page` method or I should add a separate method for building the title section of an article? (In the later case I might cut the `layout_title_page` method into some smaller parts to avoid code duplication.)
2. How should one configure the title section of an article? Should I use the values from the `title_page` section of the theme, or do you want another section (in the theme)? (In the first case, the `title_page` name is a bit misleading, while the second option would introduce duplication -- so I would opt for the first one.)

**Comment by mojavelinux** [comment]
I've already submitted a PR with this change. See #975. You're, of course, welcome to improve on it. Since parts are not allowed in the article doctype, I decided to reuse the heading_h1 category in the theme to style the doctitle. It's not perfect, but it gives us the behavior described in the workaround by default. For now, I did not designate a separate method.

**Comment by ogmios-voice** [comment]
I still don't like the idea to use the `heading_h1` style for the **title** in case of an article. Moreover the subtitle, authors, and revision information are also missing/skipped now.

I'm more than happy to fix this, if you approve it. I can think of two solutions:

1. use the `title_page` section of the theme as it is now (i.e. including the `top` values): this way we can have articles w/ logo, title, subtitle, authors, and revision also in case of articles, and only the page break will be removed. The **drawback** of this solutions is that one will have to **create a new theme** / edit the default theme, to remove the default `top` value of the title (currently `45%`) when creating an article.
2. use everything from the `title_page` section but the `top` values: the logo will be also skipped this way (but this might not be a big issue for an article), but no new theme / theme editing is required.

Please let me know if you agree with any of these. Thanks.

**Comment by mojavelinux** [comment]
@ogmios-voice I meant to follow up right after I merged my initial proposal, but I've been extremely busy.

Rather than make users wait any longer for a solution, I decided to merge my proposal in order to first solve the problem simply, without needing any additional configuration. Now we're in a position where we can build on it. And I'm open to doing that. Let's start a new issue for the enhancements. Though, they will likely go in after the alpha.16 release just because we need to get that out the door.

> I still don't like the idea to use the heading_h1 style for the title in case of an article.

I agree that it's very restrictive (and possibly over coupled). I think it can serve as a useful fallback, but it shouldn't be the only way.

> use the title_page section of the theme as it is now 

I really want to avoid using the `title_page` key for this purpose. The title_page should be about the actual title page and nothing else. Otherwise, that really would be mixing things that shouldn't be mixed.

Instead, let's see if we can introduce a special subcategory for heading that covers the doctitle. This would only control a doctitle at the top of an article (not the title page). Something like `heading_doctitle`. (If it's easier, I'd even be willing to call it heading_h0 for consistency in nomenclature).

Let's continue the discussion in a new issue.
